src/Generator/TypeBuilder.php

```diff
@@ -920,6 +924,32 @@ private function buildResolveType($resolveType)
         return $resolveType;
     }
 
+    /**
+     * Builds an arrow function from a string with an expression prefix,
+     * otherwise just returns the provided value back untouched.
+     *
+     * Render example:
+     *
+     *      fn($className) => (($className = "App\\ClassName") && $value instanceof $className)
+     *
+     * @param mixed $isTypeOf
+     *
+     * @return mixed|ArrowFunction
+     */
+    private function buildIsTypeOf($isTypeOf)
+    {
+        if (EL::isStringWithTrigger($isTypeOf)) {
+            $expression = $this->expressionConverter->convert($isTypeOf);
+
+            return ArrowFunction::new()
+                ->addArguments('className')
+                ->setExpression(Literal::new($expression))
+                ->setStatic();
+        }
+
+        return $isTypeOf;
```

Comment:
Do you think it is a good idea to allow the use of all code? Without using the expression engine you could inject any possible code. Maybe we should restrict this to be an expression string or a callable?

Reply:
At the moment we are limited to the expression engine. With your suggestion a string like this
```yaml
isTypeOf: "die('ev!l')"
```
would generate code like this:
```php
$config['isTypeOf'] = fn() => die('ev!l'),
```
but this would not be possible with the expression engine.

Reply:
Ah, sorry. I misunderstood the ArrowFunction constructor.